doc/api-design.yml

```diff
@@ -0,0 +1,903 @@
+                        type: array
+                        items:
+                          type: string
+              required:
+                - title
+                - questions
+      responses:
+        '201':
+          description: Quiz created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Quiz'
+        '400':
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    get:
+      summary: Get list of quizzes
+      tags:
+        - Quizzes
+      security:
+        - bearerAuth: []
+      parameters:
+        - name: page
+          in: query
+          schema:
+            type: integer
+          description: Page number
+        - name: limit
+          in: query
+          schema:
+            type: integer
+          description: Number of items per page
+        - name: filter
+          in: query
+          schema:
+            type: string
+            enum: [own, others, all]
+          description: Filter quizzes by ownership (own, others, or all)
+        - name: difficulty
+          in: query
+          schema:
+            type: string
+            enum: [a1,a2,b1,b2,c1,c2]
+          description: Filter quizzes by difficulty
+      responses:
+        '200':
+          description: A list of quizzes
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  quizzes:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Quiz'
+                  total_count:
+                    type: integer
+                  page:
+                    type: integer
+                  total_pages:
+                    type: integer
+        '400':
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /quizzes/{id}:
+    get:
+      summary: Get quiz details
+      tags:
+        - Quizzes
+      parameters:
+        - name: id
+          in: path
+          required: true
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: Quiz details retrieved successfully
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  quiz:
+                    $ref: '#/components/schemas/Quiz'
+                  questions:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Question'
+        '404':
+          description: Quiz not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    put:
```

Comment:
Do we have to provide all questions again for this endpoint? Or else can we update questions by id

Reply:
we have to provide all info for this endpoint. later, we will implement a new endpoint specifically for editing a question.